framework/include/utils/InfixIterator.h

```diff
@@ -17,8 +17,19 @@
 
 template <class T, class charT = char, class traits = std::char_traits<charT>>
 class infix_ostream_iterator
+/**
+ * GCC9 currently hits a "no type named 'value_type'" error during build if this
+ * is removed and iterator traits are listed within the class corresponding to the
+ * C++17 standard. This preserves use of std::iterator for GCC9 and earlier.
+ */
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ <= 9)
   : public std::iterator<std::output_iterator_tag, void, void, void, void>
+#endif
 {
+#if defined(__clang__) && (__GNUC__ > 9)
```

Comment:
```suggestion
#if defined(__clang__) || (__GNUC__ > 9)
```
We want clang OR GCC > 9 here.